cloudify_azure/resources/deployment.py

```diff
@@ -152,7 +152,7 @@ def delete(self):
 def get_template(ctx, properties):
     template = properties.get('template')
     if template:
-        if isinstance(template, basestring):
+        if isinstance(template, (text_type, bytes)):
```

Comment:
Only `text_type` is what we should use here (following “Unicode sandwich” principle)

Reply:
This does not handle the input we are getting.